### PR TITLE
Add fstar module

### PR DESCRIPTION
**Pull request by tchajed**
Add a module for [F*](https://fstar-lang.org/), a language for program verification from Microsoft Research.

I roughly followed https://github.com/kyoDralliam/fstar-layer/ to get an idea of what aspects of `fstar-mode.el` should be supported but more closely mirrored the existing Coq module as far as keybindings. This way all three people that use F* and Coq won't get confused (when the other two of them switch to doom).

### CODE REVIEW

**Comment by tchajed** [comment]
@cpitclaudel might have some thoughts as the author of `fstar-mode.el`.

**Comment by cpitclaudel** [comment]
LGTM, though is there a way to put these keybindings in the upstream package, as an evil-mode config?

File: modules/lang/fstar/README.org [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#+TITLE: :lang F*
```
Comment (hlissner) [inline modules/lang/fstar/README.org]: s/F\*/fstar

File: modules/lang/fstar/README.org [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+This module adds [[https://fstar-lang.org/][F*]] support, powered by [[https://github.com/FStarLang/fstar-mode.el][fstar-mode.el]].
```
Comment (hlissner) [inline modules/lang/fstar/README.org]: Please conform to and fill out our our README template: https://github.com/hlissner/doom-emacs/blob/develop/modules/editor/file-templates/templates/org-mode/__doom-readme

If you have the `:editor file-templates` module enabled, typing `__doom-readme` then TAB will expand it.

File: modules/lang/fstar/packages.el [hunk 3]
```diff
@@ -0,0 +1,4 @@
+;; -*- no-byte-compile: t; -*-
+;;; lang/fstar.el
```
Comment (hlissner) [inline modules/lang/fstar/packages.el]: Should be

```elisp
;;; lang/fstar/packages.el
```

**Comment by hlissner** [comment]
Looks good to me. Thanks for the PR!

**Pull request merged**